### PR TITLE
Makes head roles not sometimes fail to roll

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Makes head roles not sometimes fail to roll
If a lot of people have head roles on high startshift (Like a LOT) and RNG is bad, there's a chance people who should roll head roles won't properly roll it. Most of the time this just caused a bunch of bogus errors instead of any real issues though
Also just generally cleans up the code. We gotta refactor jobs at some point

## Why It's Good For The Game
Bugs bad

## Testing
Joined as a tot twice, once with a 100% chance of getting my head role and a 0% chance of getting a head role. worked both times
## Changelog
:cl:
fix: Fixed a rare case of head roles fizzling and not being assigned
/:cl:

### CODE REVIEW

**Review by S34NW — APPROVED** [review]

File: code/controllers/subsystem/SSjobs.dm [hunk 1]
```diff
@@ -201,15 +204,9 @@ SUBSYSTEM_DEF(jobs)
 		if(player.mind && (job.title in player.mind.restricted_roles))
 			Debug("GRJ incompatible with antagonist role, Player: [player], Job: [job.title]")
 			continue
-		if(player.mind && (job.title in SSticker.mode.single_antag_positions))
-			if(!prob(probability_of_antag_role_restriction))
-				Debug("Failed probability of getting a second antagonist position in this job, Player: [player], Job:[job.title]")
-				continue
-			else
-				probability_of_antag_role_restriction /= 10
```
Comment (SteelSlayer) [inline code/controllers/subsystem/SSjobs.dm:209]: Are you removing this and the same block in the proc below because it's redundant?

Comment (GDNgit) [reply code/controllers/subsystem/SSjobs.dm:209]: redundant now because we directly check at giving them the role

**Review by S34NW — CHANGES_REQUESTED** [review]
This has made things worse, but due to current round i cannot disclose why yet

**Comment by GDNgit** [comment]
qwerty's version will be getting merged

**Pull request closed without merging**